tests/integration/multipass/test_multipass.py

```diff
@@ -0,0 +1,244 @@
+# Copyright (C) 2021 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import io
+import shutil
+import subprocess
+
+import pytest
+
+from craft_providers.multipass import Multipass
+
+from . import conftest
+
+pytestmark = pytest.mark.skipif(
+    shutil.which("multipass") is None, reason="multipass not installed"
```

Comment:
Shouldn't we force the developer to have multipass installed and always test these?

Reply:
My thoughts:
- not all providers will support all platforms (e.g. lxd on Windows)
- not all hosts running the tests will have the capability to support the provider (e.g. Ubuntu 20.04 host without virtualization capabilities, or already running in an LXC container).

By simply keying off whether or not the developer installed it, we can reasonably determine which tests to skip automatically with the given rationale.  It's easy enough for the developer to determine, and doesn't require any magic knowledge in the test logic.